src/components/generic-filter.js

```diff
@@ -54,13 +54,199 @@ const equipmentsDefinition = {
+    },
+    BATTERY: {
+        label: 'Batteries',
+        fields: {
+            countries: {
+                name: 'Countries',
+                type: filteredTypes.countries,
+            },
+            nominalVoltage: {
+                name: 'nominalVoltage',
+                type: filteredTypes.range,
+            },
+            substationName: {
+                name: 'substationName',
+                type: filteredTypes.string,
+            },
+        },
+    },
+    SHUNT_COMPENSATOR: {
+        label: 'ShuntCompensators',
+        fields: {
+            countries: {
+                name: 'Countries',
+                type: filteredTypes.countries,
+            },
+            nominalVoltage: {
+                name: 'nominalVoltage',
+                type: filteredTypes.range,
+            },
+            substationName: {
+                name: 'substationName',
+                type: filteredTypes.string,
+            },
+        },
+    },
+    STATIC_VAR_COMPENSATOR: {
+        label: 'StaticVarCompensators',
+        fields: {
+            countries: {
+                name: 'Countries',
+                type: filteredTypes.countries,
+            },
+            nominalVoltage: {
+                name: 'nominalVoltage',
+                type: filteredTypes.range,
+            },
+            substationName: {
+                name: 'substationName',
+                type: filteredTypes.string,
+            },
+        },
+    },
+    DANGLING_LINE: {
+        label: 'DanglingLines',
+        fields: {
+            countries: {
+                name: 'Countries',
+                type: filteredTypes.countries,
+            },
+            nominalVoltage: {
+                name: 'nominalVoltage',
+                type: filteredTypes.range,
+            },
+            substationName: {
+                name: 'substationName',
+                type: filteredTypes.string,
+            },
+        },
+    },
+    LCC_CONVERTER_STATION: {
+        label: 'LccConverterStations',
+        fields: {
+            countries: {
+                name: 'Countries',
+                type: filteredTypes.countries,
+            },
+            nominalVoltage: {
+                name: 'nominalVoltage',
+                type: filteredTypes.range,
+            },
+            substationName: {
+                name: 'substationName',
+                type: filteredTypes.string,
+            },
+        },
+    },
+    VSC_CONVERTER_STATION: {
+        label: 'VscConverterStations',
+        fields: {
+            countries: {
+                name: 'Countries',
+                type: filteredTypes.countries,
+            },
+            nominalVoltage: {
+                name: 'nominalVoltage',
+                type: filteredTypes.range,
+            },
+            substationName: {
+                name: 'substationName',
+                type: filteredTypes.string,
+            },
+        },
+    },
 };
 
+function deepCopy(aObject) {
```

Comment:
Here, we cannot use spread operator, because we really need a deep copy.
This is a correction of an existing bug in the master, when we define a line filter and specify at the same time both nominal voltage 1 and nominal voltage 2 criteria : the nominal voltage 1 criteria values are overriden by the nominal voltage 2 criteria values ...